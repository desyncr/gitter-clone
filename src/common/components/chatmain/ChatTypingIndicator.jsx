import React from 'react';

const ChatTypingIndicator = ({ usersTyping }) => {
  let indicatorMessage;
  if (usersTyping.length === 1) {
    indicatorMessage = usersTyping[0] + ' is typing';
  } else if (usersTyping.length <= 3) {
    indicatorMessage = usersTyping.join(' and ') + ' are typing';
  } else {
    indicatorMessage = usersTyping.length + ' users are typing';
  }
  return (
    <div>
      {indicatorMessage}
    </div>
  );
};

ChatTypingIndicator.propTypes = {
  usersTyping: React.PropTypes.array.isRequired
};

export default ChatTypingIndicator;
