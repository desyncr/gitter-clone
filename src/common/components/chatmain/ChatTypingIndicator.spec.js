import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ChatTypingIndicator from './ChatTypingIndicator';

const setup = () => {
  const props = {
    isTyping: 'Dan'
  };
  const component = <ChatTypingIndicator {...props} />;

  return {
    props,
    component
  };
};

describe('<ChatTypingIndicator />', () => {
  it('should render correctly', () => {
    const { component } = setup();
    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
