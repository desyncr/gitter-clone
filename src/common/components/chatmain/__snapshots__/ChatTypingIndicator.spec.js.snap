// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatTypingIndicator /> should render correctly with >3 users typing 1`] = `
<div>
  4 users are typing
</div>
`;

exports[`<ChatTypingIndicator /> should render correctly with 1 user typing 1`] = `
<div>
  Dan is typing
</div>
`;

exports[`<ChatTypingIndicator /> should render correctly with 2 users typing 1`] = `
<div>
  Dan and Liz are typing
</div>
`;

exports[`<ChatTypingIndicator /> should render correctly with 3 users typing 1`] = `
<div>
  Dan and Liz and Joe are typing
</div>
`;
